lib/toolshed/http.ex

```diff
@@ -67,6 +67,92 @@ defmodule Toolshed.HTTP do
     end
   end
 
+  @doc """
+  Perform a HTTP GET request for the specified URL
+
+  By default, the results are printed or you can optionally choose to download
+  it to a specific location on the file system.
+
+  Options:
+
+  * `:dest` - File path to write the response to. Defaults to printing to the terminal.
+  * `:verbose` - Display request and response headers. Disabled by default.
+  """
+  def httpget(url, options \\ []) do
+    check_inets()
+
+    url = url_defaults(url)
+    dest = Keyword.get(options, :dest, nil)
+    verbose = Keyword.get(options, :verbose, false)
+
+    stream =
+      if dest != nil do
+        to_charlist(dest)
+      else
+        :self
+      end
+
+    request_headers = [{'User-Agent', 'curl'}]
+
+    if verbose do
+      display_headers(request_headers, ">")
+    end
+
+    Task.async(fn ->
+      {:ok, _ref} =
+        :httpc.request(
+          :get,
+          {to_charlist(url), request_headers},
+          [],
+          sync: false,
+          stream: stream
+        )
+
+      handle_stream(verbose)
+    end)
+    |> Task.await()
+
+    IEx.dont_display_result()
+  end
+
+  defp url_defaults(url) do
+    case URI.parse(url) do
+      %URI{scheme: nil} -> "http://#{url}"
+      _ -> url
+    end
+  end
+
+  defp display_headers(headers, prefix) do
+    Enum.each(headers, fn {k, v} ->
+      IO.puts(:stderr, "#{prefix} #{k}: #{v}")
+    end)
+  end
+
+  defp handle_stream(verbose) do
+    receive do
+      {:http, {_ref, :stream_start, headers}} ->
+        if verbose do
+          display_headers(headers, "<")
+        end
+
+        handle_stream(verbose)
+
+      {:http, {_ref, :stream, body}} ->
+        IO.write(body)
```

Comment:
```suggestion
        IO.binwrite(body)
```

I just tried this out by running `httpget "google.com"` and the response I got wasn't Unicode, so `IO.write/1` raised. It seems like an easy fix is to change this to `IO.binwrite/1`. The other option would be handle the encoding based on the response header. Do you have opinions? If you're ok with `IO.binwrite/1`, I'll make the change and squash everything together and merge.

Reply:
@fhunleth Curl doesn't appear to do any charset decoding so I'm fine with using `IO.binwrite` as this command is in the same spirit. Although [the docs](https://hexdocs.pm/elixir/IO.html#binwrite/2) for `binwrite` make it sound like this might not work as expected? Specifically the note at the bottom.

Reply:
Yeah, that note sounds scary, but I've been trying out non-Unicode-encoded sites and it seems fine. 

I went ahead and manually merged your PR with the change to `binwrite` and the spelling fixes. Thanks!!